pkg/v1/providers/infrastructure-aws/v1.2.0/yttcc/overlay.yaml

```diff
@@ -87,15 +88,16 @@ spec:
         #@ if data.values.NODE_MACHINE_TYPE_2:
         variables:
           overrides:
-          - name: NODE_MACHINE_TYPE
-            value: #@ data.values.NODE_MACHINE_TYPE_2
+          - name: worker
+            value:
+              instanceType: #@ data.values.NODE_MACHINE_TYPE_2
         #@ end
       #@ end
     #@overlay/match missing_ok=True
     variables:
-    #@ vars = get_cluster_variables()
+    #@ vars = get_aws_vars()
     #@ for configVariable in vars:
-    #@  if vars[configVariable] != None and configVariable in ["AWS_REGION", "AWS_SSH_KEY_NAME", "BASTION_HOST_ENABLED", "AWS_VPC_CIDR", "AWS_VPC_ID", "AWS_NODE_AZ", "AWS_PRIVATE_NODE_CIDR", "AWS_PRIVATE_SUBNET_ID", "AWS_PUBLIC_NODE_CIDR", "AWS_PUBLIC_SUBNET_ID", "CONTROL_PLANE_MACHINE_TYPE", "NODE_MACHINE_TYPE", "CLUSTER_NAME", "AWS_CONTROL_PLANE_OS_DISK_SIZE_GIB", "AWS_NODE_OS_DISK_SIZE_GIB", "AWS_SECURITY_GROUP_BASTION", "AWS_SECURITY_GROUP_CONTROLPLANE", "AWS_SECURITY_GROUP_APISERVER_LB", "AWS_SECURITY_GROUP_NODE", "AWS_SECURITY_GROUP_LB", "AWS_LOAD_BALANCER_SCHEME_INTERNAL", "AWS_IDENTITY_REF_KIND", "AWS_IDENTITY_REF_NAME", "CLUSTER_PLAN", "proxy", "TKG_PROXY_CA_CERT", "TKG_CUSTOM_IMAGE_REPOSITORY", "TKG_CUSTOM_IMAGE_REPOSITORY_SKIP_TLS_VERIFY", "TKG_CUSTOM_IMAGE_REPOSITORY_CA_CERTIFICATE", "ENABLE_AUDIT_LOGGING"]:
+    #@  if vars[configVariable] != None and configVariable in ["region", "sshKeyName", "bastion", "network", "controlPlane", "worker", "loadBalancerSchemeInternal", "identityRef", "imageRepository", "trust", "auditLogging"]:
```

Comment:
question: if the purpose of get_aws_vars is to construct the appropriate set of variables specific to aws, do we still need this check? (same for other infras as well)

Reply:
It is impractical to remove this at the moment. The optional_params in our tests will create test files that mix and match aws and vsphere properties. When those get passed through as clusterclass variables there are errors when a variable is added which doesn't have a corresponding variable definition in the clusterclass. The current implementation allows us to filter these variables. I'm very open to setting up a new issue to address problems with our clustergen tests generation.